src/components/CandidateForm.jsx

```diff
@@ -0,0 +1,40 @@
+import React, { useState, useEffect } from "react";
+
+import styles  from "./CandidateForm.module.css";
+import Button from "./Button";
+
+const CandidateForm = ({ inputs, handleSubmission }) => {
+    //const [data, setData] = useState(inputs.reduce((data, val) => { return {...data, [val.name]: val}}, {}))
+    const [data, setData] = useState(inputs.reduce((data, val) => { return {...data, [val.name] : val.value}}, {}))
+
+    const onFormSubmission = (event) => {
+        event.preventDefault()
+        handleSubmission(data)
+    } 
+
+    return (
+        <>
+        <form className={styles.form}>
+            <fieldset className={styles.field}>
+                {
+                    inputs.map(item => {
+                        return (
+                            <input 
+                                className={styles.field}
+                                key={item.name}
+                                name={item.name}
+                                placeholder={item.placeholder}
```

Comment:
With the first comment in mind, now your placeholder, key, and name depend on props, while the value and on change depend on the local state data. 

This creates 2 sources of truth and its can get really hard to maintain.